src/client/Client.h

```diff
@@ -909,9 +939,10 @@ class Client : public Dispatcher, public md_config_obs_t {
   void handle_client_reply(const MConstRef<MClientReply>& reply);
   bool is_dir_operation(MetaRequest *request);
 
-  int path_walk(const filepath& fp, struct walk_dentry_result* result, const UserPerm& perms, bool followsym=true, int mask=0);
+  int path_walk(const filepath& fp, struct walk_dentry_result* result, const UserPerm& perms, bool followsym=true, int mask=0,
+                InodeRef dirinode=nullptr);
```

Comment:
```suggestion
  int path_walk(InodeRef dirinode=nullptr, const filepath& fp, struct walk_dentry_result* result, const UserPerm& perms, bool followsym=true, int mask=0);
```

Reply:
Well, only trailing arguments can have default values. Unless you recommend another path_walk() decl that calls this with dirinode as nullptr?

Reply:
Ah, nvm.